### PR TITLE
Remove metadata keys with null values

### PR DESCRIPTION
**Pull request by hakontro**
This was already done for insertInto, but is now done for all ways of
writing to CDF.

### CODE REVIEW

**Comment by hakontro** [comment]
```
rows.map { r =>
  val items = fromRow[ResourceItem](r)
  items.copy(metadata = filterMetadata(items.metadata))
}
```
Couldn't find a good way to abstract over this pattern. Writing a function like `def getFromRowAndFilterMetadata[A <: HasMetadata](rows: Seq[Row])` is tricky, since the copy-method is auto-generated for case classes. Anyone see a good way?

File: src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala [hunk 1]
```diff
@@ -120,7 +120,10 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
 
   override def update(rows: Seq[Row]): IO[Unit] = {
     val updateAssetsItems = rows
-      .map(r => fromRow[UpdateAssetsItemBase](r))
+      .map { r =>
+        val assetItems = fromRow[UpdateAssetsItemBase](r)
```
Comment (wjoel) [inline src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala]: `assetItem`

Comment (hakontro) [reply src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala]: True! Found a few more of those.

**Comment by hakontro** [comment]
@wjoel Good catch, new tests caught some bugs.

**Pull request merged**